src/components/MultiHopTrade/components/TradeAmountInput.tsx

```diff
@@ -58,7 +59,7 @@ export type TradeAmountInputProps = {
   assetSymbol: string
   assetIcon: string
   onChange?: (value: string, isFiat?: boolean) => void
-  onMaxClick?: () => Promise<void>
+  onMaxClick?: (isFiat: boolean) => Promise<void>
```

Comment:
can we just make this required for this component and omit from the `TradeAssetInput` props if we are just defining `onMaxClick` locally there?

Reply:
this also fixes the later type narrowing issues

Reply:
Much better that way, and indeed, since for this specific component, `onMaxClick` is effectively required, and always defined since we define it ourselves in `<TradeAssetInput />` there's no reason why we wouldn't make it required at type-level 🙏🏽 :
https://github.com/shapeshift/web/pull/5961/files#r1446451716